x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/CcrAutoFollowInfoFetcher.java

```diff
@@ -0,0 +1,140 @@
+            listener.map((clusterAutoFollowedSystemIndices) ->
+                clusterAutoFollowedSystemIndices.stream()
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toList())),
+            remoteClusterAutoFollowedIndices.size()
+        );
+
+        // When a system index is followed we don't copy over the isSystem flag into IndexMetadata
+        // We need to fetch the remote cluster state in order to check whether or not the following index
+        // follows a leader system index.
+        for (Map.Entry<String, List<AutoFollowedIndex>> remoteAutoFollowedIndicesEntry : remoteClusterAutoFollowedIndices.entrySet()) {
+            final String remoteCluster = remoteAutoFollowedIndicesEntry.getKey();
+            final List<AutoFollowedIndex> autoFollowedIndices = remoteAutoFollowedIndicesEntry.getValue();
+
+            try {
+                client.getRemoteClusterClient(remoteCluster)
+                    .admin()
+                    .cluster()
+                    .prepareState()
+                    .clear()
+                    .setMetadata(true)
+                    .execute(new ActionListener<ClusterStateResponse>() {
+                        @Override
+                        public void onResponse(ClusterStateResponse stateResponse) {
+                            final ClusterState clusterState = stateResponse.getState();
+                            List<String> autoFollowedSystemIndices = new ArrayList<>();
+                            for (AutoFollowedIndex autoFollowedIndex : autoFollowedIndices) {
+                                final IndexAbstraction indexAbstraction =
+                                    clusterState.metadata().getIndicesLookup().get(autoFollowedIndex.remoteIndexName);
+                                if (indexAbstraction != null && indexAbstraction.isSystem()) {
+                                    autoFollowedSystemIndices.add(autoFollowedIndex.localIndexName);
+                                }
+                            }
+                            clusterResponsesListener.onResponse(autoFollowedSystemIndices);
+                        }
+
+                        @Override
+                        public void onFailure(Exception e) {
+                            clusterResponsesListener.onFailure(e);
+                        }
+                    });
+            } catch (IllegalArgumentException e) {
+                clusterResponsesListener.onFailure(e);
+            }
+        }
+    }
+
+    private static boolean isCurrentlyFollowed(PersistentTasksCustomMetadata persistentTasks, String indexName) {
+        return persistentTasks != null && persistentTasks.findTasks(ShardFollowTask.NAME, task -> true).stream()
+            .map(task -> (ShardFollowTask) task.getParams())
+            .anyMatch(shardFollowTask -> indexName.equals(shardFollowTask.getFollowShardId().getIndexName()));
```

Comment:
Perhaps use `Index` rather than the index name, so that we compare both uuid and name.

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/73237/commits/b80e0e99d1f83f10afc0cbce0f85e724d92dd9b0